### PR TITLE
New dataset and pipeline card layout

### PR DESCRIPTION
**Pull request by cmadjar**
The dataset and pipeline cards have been redesigned in this PR as discussed in the past few weeks during the CONP dev calls.

Changes include:
- refactoring of the download and process columns into buttons:
  - one button for direct download
  - one button for the link to the DataLad instructions for download
  - one button to process on CBRAIN
- addition of circled question mark with tooltips next to the following features to provide users with more explanation of the feature:
  - for the "Direct Download" button
  - for the "Download With DataLad" button
  - for the "Process On CBRAIN" button
  - for the "Metadata file" download link
- harmonization of the look of the dataset and pipeline cards
- moved the download metadata into the detailed page

New layout for a dataset:
Screen Shot 2022-05-11 at 5 40 58 PM

New card layout for a tool:
Screen Shot 2022-05-11 at 5 41 21 PM

### CODE REVIEW

**Comment by cmadjar** [comment]
@tkkuehn I am still waiting for feedbacks from Patrick about the text of the tooltips which is why I marked the PR as blocked but maybe you could review the rest and test in the meantime?

Thank you!!

File: app/static/lib/conp-react/src/DatasetElement/index.js [hunk 1]
```diff
@@ -48,250 +47,259 @@ const DatasetElement = (props) => {
...
+                >
+                  Direct Download ({element.size})
+                </button> :
+                <div className="btn btn-outline-secondary m-1 disabled">
+                  Direct Download (Not Available)
+                </div>
+              }
+            </div>
+            <div className="col-2 p-2">
+              <p className="card-text pl-1">
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/DatasetElement/index.js:196]: Maybe make this a `div`? React complains for me in the dev console about having a `ReactToolTip` as a descendant of a `p`.

File: app/static/lib/conp-react/src/DatasetElement/index.js [hunk 2]
```diff
@@ -48,250 +47,259 @@ const DatasetElement = (props) => {
...
-                    {metadataErrorText}
+          <div className="row align-items-center w-100">
+            <div className="col-10 p-0">
+              <a href={`dataset?id=${element.id}#dataladInstructions`} role="button"
+                   className="btn btn-success m-1">
+                Download With DataLad
+              </a>
+            </div>
+            <div className="col-2 p-2">
+              <p className="card-text pl-1">
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/DatasetElement/index.js:225]: Maybe make this a `div`? React complains for me in the dev console about having a `ReactToolTip` as a descendant of a `p`.

File: app/static/lib/conp-react/src/DatasetElement/index.js [hunk 3]
```diff
@@ -48,250 +47,259 @@ const DatasetElement = (props) => {
...
-                <div className="d-flex">
-                  {authIcons.map((icon, index) => <div key={"authIcon_" + index}
-                                                       className="text-center p-1">{icon}</div>)}
-                </div>
-              </div>
+                </button>
+              }
+            </div>
+            <div className="col-2 p-2">
+              <p className="card-text pl-1">
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/DatasetElement/index.js:266]: Maybe make this a `div`? React complains for me in the dev console about having a `ReactToolTip` as a descendant of a `p`.

File: app/static/lib/conp-react/src/PipelineElement/index.js [hunk 4]
```diff
@@ -17,135 +22,121 @@ const PipelineElement = (props) => {
...
-          data-toggle="tooltip"
-          title="Unavailable"
-          style={{ maxWidth: "140px" }}
-        >
-          <a className="btn disabled" href={item.uri} disabled>
-            <img className="img-fluid" alt="Online platform" src={item.img} />
-          </a>
-        </span>
+      <div className="col-2 p-2">
+        <p className="card-text pl-1">
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/PipelineElement/index.js:56]: Maybe change this to `div`.

File: app/static/lib/conp-react/src/PipelineElement/index.js [hunk 5]
```diff
@@ -17,135 +22,121 @@ const PipelineElement = (props) => {
...
-            </button>
-          }
-        </span>
+    <div className="row w-100 align-items-center">
+      <div className="col-10 p-0">
+        {item.uri ?
+          <button className="btn btn-outline-success m-1" onClick={openCbrainModal}>
+            <div className="d-flex row align-items-center justify-content-center">
+              Use This Tool On <img
+              className="cbrain-img justify-content-center align-items-center pl-4"
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/PipelineElement/index.js:40]: Add something like `alt="CBRAIN logo"`

File: app/static/lib/conp-react/src/PipelineElement/index.js [hunk 6]
```diff
@@ -17,135 +22,121 @@ const PipelineElement = (props) => {
...
+              src="static/img/cbrain-long-logo-blue.png" style={{maxHeight: '30px'}}
+            />
+            </div>
+          </button> :
+          <button className="btn btn-outline-secondary disabled m-1">
+            <div className="d-flex row align-items-center justify-content-center">
+              Use This Tool On
+              <img
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/PipelineElement/index.js:48]: Add something like `alt="Disabled CBRAIN logo"`

File: app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    $("#downloadModal").modal("hide");
+  };
+
+  return ReactDOM.createPortal(
+    <div className="modal-content">
+      <div className="modal-body">
+        <table>
+         <tr>
+           <td style={{width:"150px"}}>
+             <img src="static/img/conp.png" className="text-center" style={{width:"100%"}}/>
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js:18]: Add something like `alt="CONP logo"`

File: app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js:1]: `useState` isn't used.

File: app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+import * as ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+
+const DownloadModalWindowElement = (props) => {
+  const { authorized, imagePath, ...element } = props;
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js:6]: I think only `element.size` and `element.zipLocation` are used, so we could just pass those two directly and use a lot fewer props for this component.

File: app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+           Download
+         </a>
+       </div>
+     </div>
+    </div>,
+    document.querySelector("#downloadModalDocument")
+  );
+};
+
+DownloadModalWindowElement.propTypes = {
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js:55]: Again, we could get rid of most of these props.

File: app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+  const { authorized, imagePath, ...element } = props;
+
+  const close = (event) => {
+    $("#downloadModal").modal("hide");
+  };
+
+  return ReactDOM.createPortal(
+    <div className="modal-content">
+      <div className="modal-body">
+        <table>
```
Comment (tkkuehn) [inline app/static/lib/conp-react/src/DatasetElement/DownloadModalWindowElement/index.js:15]: I would rather use a bootstrap container for laying out the CONP image and warning text if we could.

**Review by tkkuehn — CHANGES_REQUESTED** [review]
A few relatively minor changes but this overall looks great, thanks for putting it together!

**Comment by cmadjar** [comment]
@tkkuehn thank you for all the great suggestions! I should have addressed all of them. Ready for another look and test :-).

FYI - the tooltip texts have also been updated.

Thanks again!!

**Review by tkkuehn — APPROVED** [review]

**Pull request merged**